scripts/scr_marine_struct/scr_marine_struct.gml

```diff
@@ -614,6 +614,22 @@ function TTRPG_stats(faction, comp, mar, class = "marine") constructor{
 		return obj_ini.role[company][marine_number];
 	};
 	static update_role = function(new_role){
+		if (base_group=="astartes"){
+			if (role() == obj_ini.role[100][12] && new_role!=obj_ini.role[100][12]){
+		  		var has_carpace =false;
+		  		if (struct_exists(body[$ "torso"], "black_carpace")){
```

Comment:
wouldn't it be better to have a separate `add_gene_seed_organ(...)`, then on init/update_role marines would get the organs along with any effects they grant, while other races wouldn't?

Reply:
for now, just a separate method(s) to update gene_seed stuff, will be clearer and easier to maintain

Reply:
> (snip) it might be better to hold the black carapace on the body struct that way we can make it applicable as a mechanic for putting "powered armour" on orks or enhaneced armour on sisters (with the general idea being the black_carapace is simply the area to hold powered or upgraded armour for a unit depending on it's class) with "Black Caraoace" simply being the display name for marines

hmm, interesting idea, but we need to discuss/brainstorm this one. I see a danger of splitting gene seed code all over different areas, which is not something that we want. I'd go either with just keeping body and geneseed separate OR integrating them 100%. I'll create a thread on Discord